diff --git a/node_modules/@remax/cli/lib/build/ProjectPath.js b/node_modules/@remax/cli/lib/build/ProjectPath.js
index dd48203..61bbf87 100644
--- a/node_modules/@remax/cli/lib/build/ProjectPath.js
+++ b/node_modules/@remax/cli/lib/build/ProjectPath.js
@@ -70,6 +70,12 @@ class ProjectPath {
             if (fs.existsSync(extFile)) {
                 return shared_1.slash(extFile);
             }
+            const reg = path.sep === "/" ? /packages\/([^/]*)\/src/ : /packages\\([^/]*)\\src/
+            const otherExtFile = extFile.replace(reg, `packages${path.sep}commonHis${path.sep}src`)
+            if (fs.existsSync(otherExtFile)) {
+                return shared_1.slash(otherExtFile);
+            }
+
         }
         return '';
     }
